File: engine/operations/delete.py
"""会话删除用例:永久删除,不留恢复快照。"""
from __future__ import annotations

from ..context import EngineContext
from ..errors import require_agent_capability


class SessionDeletionService:
    def __init__(self, ports: EngineContext):
        self._ports = ports

    def delete(self, tool: str, reference: str) -> dict:
        adapter = self._ports.adapter(tool)
        lifecycle = require_agent_capability(adapter, "delete", "lifecycle")
        return lifecycle.delete(adapter, reference)

File: crates/ferry-engine/src/operations/delete.rs
//! 会话删除用例：永久删除，不留恢复快照。
//!
//! 语义事实源：`engine/operations/delete.py`。

use serde_json::{Map, Value};

use crate::operations::types::{EngineResult, Ports};

pub struct SessionDeletionService {
    ports: Ports,
}

impl SessionDeletionService {
    pub fn new(ports: Ports) -> Self {
        Self { ports }
    }

    /// `lifecycle.delete(adapter, reference)`；能力门禁走 `delete`。
    pub fn delete(&self, tool: &str, reference: &str) -> EngineResult<Map<String, Value>> {
        let adapter = self.ports.adapter(tool)?;
        let lifecycle = adapter.require_lifecycle("delete")?;
        Ok(lifecycle.delete(&adapter, reference)?)
    }
}
